### ISSUE TITLE
perf: chunk reranker input into CQS_RERANKER_BATCH-sized ORT runs

### ISSUE THREAD

**Issue by jamie8johnson** [jamie8johnson/cqs#963]
## Problem

The reranker tokenizes and runs all `(query, passage)` pairs in a single ORT `session.run` call at `src/reranker.rs:150-210`. With `k=100` and `max_length=512` that is a `[100, 512]` token tensor plus matching mask and type tensors.

On a small GPU, or after SPLADE has already claimed VRAM for its session, this single-shot batch OOMs. `--rerank-top 500` with a cross-encoder walks straight into a 500-pair run.

There is no configurable batch cap. Compare with the embed path at `src/embedder/mod.rs:544-562`, which already honors `CQS_EMBED_BATCH_SIZE` and splits large inputs into several `session.run` calls.

## Proposal

Split the reranker input into batches of `CQS_RERANKER_BATCH` pairs. Default 32 or 64 (pick based on A6000 / 4000 VRAM headroom). Mirror the `embed_documents` pattern:

```rust
for chunk in pairs.chunks(batch_size) {
    let logits = session.run(chunk)?;
    scores.extend(logits);
}
```

Each sub-batch produces its own tensor allocation that ORT releases before the next iteration. Peak VRAM drops from `O(n * max_length)` to `O(batch_size * max_length)`.

When `CQS_RERANKER_BATCH` is unset, auto-pick based on available VRAM at session init (or fall back to a safe 32).

## Acceptance criteria

- [ ] `CQS_RERANKER_BATCH` env var plumbed through `Reranker::predict` (or equivalent)
- [ ] Default batch size safe for 8GB GPUs (matches the small-GPU profile in MEMORY.md)
- [ ] Reranker with `--rerank-top 500` completes on an 8GB GPU without OOM
- [ ] Unit test that batching is observable (session.run call count = ceil(n / batch_size))
- [ ] README env-var table updated

## Cross-references

- Audit triage: `docs/audit-triage.md` § "P4 GitHub issues (file separately)" — SHL-V1.25-8
- Audit finding: `docs/audit-findings.md` § SHL-V1.25-8 (line 575)
- Precedent: `CQS_EMBED_BATCH_SIZE` in `src/embedder/mod.rs:544-562`
